### PR TITLE
Added purecomponent for storage dashboard

### PR DESCRIPTION
**Pull request by cloudbehl**

### CODE REVIEW

File: src/components/StorageOverview/Events/Events.js [hunk 1]
```diff
@@ -12,26 +12,43 @@ import {
...
+  }
+}
+
+Events.defaultProps = {
+  EventStreamComponent: React.Fragment,
+};
 
 Events.propTypes = {
-  EventStreamComponent: PropTypes.oneOfType([PropTypes.node, PropTypes.func]).isRequired,
+  EventStreamComponent: PropTypes.oneOfType([PropTypes.node, PropTypes.func]),
```
Comment (mareklibra) [inline src/components/StorageOverview/Events/Events.js]: Why this change?

File: src/components/StorageOverview/Events/Events.js [hunk 2]
```diff
@@ -12,26 +12,43 @@ import {
...
+    const { EventStreamComponent } = this.props;
+    return (
+      <DashboardCard>
+        <DashboardCardHeader className="kubevirt-events__card-header">
+          <DashboardCardTitle>Events</DashboardCardTitle>
+          <DashboardCardTitleHelp>help for events</DashboardCardTitleHelp>
+        </DashboardCardHeader>
+        <DashboardCardBody id="events-body" className="kubevirt-events__card-body">
+          <EventsBody>
+            <EventStreamComponent />
```
Comment (mareklibra) [inline src/components/StorageOverview/Events/Events.js]: With the change bellow, the `EvenStreamComponent` can be `undefined` here.

File: src/components/StorageOverview/Events/tests/__snapshots__/Events.test.js.snap [hunk 3]
```diff
@@ -22,9 +22,7 @@ exports[`<Events /> renders correctly 1`] = `
     id="events-body"
     isLoading={false}
   >
-    <EventsBody>
-      <Component />
-    </EventsBody>
+    <EventsBody />
```
Comment (mareklibra) [inline src/components/StorageOverview/Events/tests/__snapshots__/Events.test.js.snap]: I guess this is not the flow you intended to test. Can you please modify the tests to pass relevant data?

File: src/components/StorageOverview/Details/tests/__snapshots__/Details.test.js.snap [hunk 4]
```diff
@@ -20,7 +20,7 @@ exports[`<StorageDetails /> renders Loading storage details correctly 1`] = `
     <DetailsBody>
       <DetailItem
         LoadingComponent={[Function]}
-        isLoading={true}
+        isLoading={false}
```
Comment (mareklibra) [inline src/components/StorageOverview/Details/tests/__snapshots__/Details.test.js.snap]: Are relevant data passed to the test?

**Comment by cloudbehl** [comment]
@mareklibra please review.

**Comment by gnehapk** [comment]
@cloudbehl Build failed. Can you please check.

**Comment by cloudbehl** [comment]
@rawagner please review
